src/storage/admin/StatisTask.cpp

```diff
@@ -0,0 +1,229 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "kvstore/Common.h"
+#include "storage/admin/StatisTask.h"
+#include "utils/NebulaKeyUtils.h"
+
+namespace nebula {
+namespace storage {
+
+cpp2::ErrorCode
+StatisTask::getSchemas(GraphSpaceID spaceId) {
+    CHECK_NOTNULL(env_->schemaMan_);
+    auto tags = env_->schemaMan_->getAllVerTagSchema(spaceId);
+    if (!tags.ok()) {
+        return cpp2::ErrorCode::E_SPACE_NOT_FOUND;
+    }
+
+    auto edges = env_->schemaMan_->getAllVerEdgeSchema(spaceId);
+    if (!edges.ok()) {
+        return cpp2::ErrorCode::E_SPACE_NOT_FOUND;
+    }
+
+    for (auto tag : tags.value()) {
+        tags_.emplace_back(tag.first);
+    }
+    for (auto edge : edges.value()) {
+        edges_.emplace_back(edge.first);
+    }
+    return cpp2::ErrorCode::SUCCEEDED;
+}
+
+ErrorOr<cpp2::ErrorCode, std::vector<AdminSubTask>>
+StatisTask::genSubTasks() {
+    spaceId_ = ctx_.parameters_.space_id;
+    auto parts = ctx_.parameters_.parts;
+    subTaskSize_ = parts.size();
+
+    auto ret = getSchemas(spaceId_);
+    if (ret != cpp2::ErrorCode::SUCCEEDED) {
+        LOG(ERROR) << "Space not found, spaceId: " << spaceId_;
+        return ret;
+    }
+
+    std::vector<AdminSubTask> tasks;
+    for (const auto& part : parts) {
+        std::function<kvstore::ResultCode()> task = std::bind(&StatisTask::genSubTask,
+                                                              this, spaceId_, part,
+                                                              tags_, edges_);
+        tasks.emplace_back(std::move(task));
+    }
+    return tasks;
+}
+
+// Statis the specified tags and edges
+kvstore::ResultCode
+StatisTask::genSubTask(GraphSpaceID spaceId,
+                       PartitionID part,
+                       std::vector<TagID> tags,
+                       std::vector<EdgeType> edges) {
+    auto vIdLenRet = env_->schemaMan_->getSpaceVidLen(spaceId);
+    if (!vIdLenRet.ok()) {
+        LOG(ERROR) << "Get space vid length failed";
+        return kvstore::ResultCode::ERR_SPACE_NOT_FOUND;
+    }
+    auto vIdLen = vIdLenRet.value();
+
+    LOG(INFO) << "Start statis task";
+    CHECK_NOTNULL(env_->kvstore_);
+    auto prefix = NebulaKeyUtils::partPrefix(part);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = env_->kvstore_->prefix(spaceId, part, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
```

Comment:
You need to bypass the leader_change, even it is not leader, still read it.

Reply:
Got,  I will open a new pr to handle it.

Reply:
Fixed, thanks for your reminder！